---
title: "Transcriptomics and the analysis of RNA-Seq data"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Importing countData
```{r}
counts <- read.csv("airway_scaledcounts.csv", stringsAsFactors = FALSE)
metadata <-  read.csv("airway_metadata.csv", stringsAsFactors = FALSE)
```

take a look of the count
```{r}
head(counts)
```
```{r}
View(counts)
```

```{r}
head(metadata)
```

Look at metaData more carefully to figure out which is treated and which is control
```{r}
View(metadata)
```

If we look at our metadata, we see that the control samples are SRR1039508, SRR1039512, SRR1039516, and SRR1039520. This bit of code will first find the sample id for those labeled control. Then calculate the mean counts per gene across these samples
```{r}
control <- metadata[metadata[,"dex"]=="control",]
control.mean <- rowSums( counts[ ,control$id] )/4 
names(control.mean) <- counts$ensgene
```

```{r}
View(control.mean)
```

Q1: More robust way
```{r}
colnames(counts)
```
```{r}
metadata$id
```
check for correspondence of metadata and countdata
```{r}
counts[-1]
```

```{r}
all( colnames(counts)[-1] == metadata$id )
```

First task find which column in **count** correspond to the control conditions(e.g no drug)
```{r}
control.inds <- metadata[, "dex"] == "control"
control.name <- metadata$id[control.inds]
control1 <- counts[,control.name]
head(control1)
rowSums(control1)/ncol(control1)
```

Do the same thing for treated
```{r}
treated <- metadata[metadata[,"dex"]=="treated",]
treated.mean <- rowSums( counts[ ,treated$id] )/4 
names(treated.mean) <- counts$ensgene
```

More robust way for treated
```{r}
treated.inds <- metadata[, "dex"] == "treated"
treated.name <- metadata$id[treated.inds]
treated1 <- counts[,treated.name]
head(treated1)
treated.mean1 <- rowSums(treated1)/ncol(treated1)
names(treated.mean1) <- counts$ensgene
```

```{r}
meancounts <- data.frame(control.mean, treated.mean)
colSums(meancounts)
```

Now we can plot the drug treated vs non-drug  control count values
```{r}
plot(meancounts$control.mean,meancounts$treated.mean, xlab = "log-Control", ylab = "log-Treated",log = "xy")
```

Here we calculate log2foldchange, add it to our meancounts data.frame 
```{r}
meancounts$log2fc <- log2(meancounts[,"treated.mean"]/meancounts[,"control.mean"])
head(meancounts)
```

```{r}
toy <- c(1,2,0,4,0)
toy == 0
which(toy == 0)
```
```{r}
toy <- meancounts[1:6,]
toy[,1:2]
toy[,1:2]==0
which(toy[,1:2] == 0, arr.ind = TRUE)
unique(which(toy[,1:2] == 0, arr.ind = TRUE)[,1])
```

Letâ€™s filter our data to remove these genes.
```{r}
zero.vals <- which(meancounts[,1:2]==0, arr.ind=TRUE)

to.rm <- unique(zero.vals[,1])
mycounts <- meancounts[-to.rm,]
head(mycounts)
```

```{r}
up.ind <- mycounts$log2fc > 2
down.ind <- mycounts$log2fc < (-2)
```

```{r}
summary(up.ind)
summary(down.ind)
```

There are `sum(up.ind)` up regulated genes and `sum(down.ind)` down regulated genes.


```{r}
head(mycounts[up.ind,])
```

## Adding Annotation data

```{r}
anno <- read.csv("annotables_grch38.csv")
head(anno)
```
```{r}
head(mycounts)
```

```{r}
mycounts.annotated <- merge(anno, mycounts, by.x = "ensgene", by.y = "row.names")
mycounts.annotated
```

## DESeq2 analysis

```{r}
library(DESeq2)
citation("DESeq2")
```
```{r}
dds <- DESeqDataSetFromMatrix(countData=counts, 
                              colData=metadata, 
                              design=~dex, 
                              tidy=TRUE)
dds
```

```{r}
dds <- DESeq(dds)
res <- results(dds)
res
```

```{r}
summary(res)
```

```{r}
res05 <- results(dds, alpha = 0.05)
summary(res05)
```
```{r}
res01 <- results(dds, alpha = 0.05)
summary(res01)
```

Volcano Plot
```{r}
mycols <- rep("gray", nrow(res01))
mycols[abs(res01$log2FoldChange) > 2] <- "red"
inds <- (res01$padj < 0.01) & (abs(res01$log2FoldChange) > 2)
mycols[inds] <- "blue"
plot(res01$log2FoldChange, -log(res01$padj), col = mycols, xlab = "Log2(FoldChange)", ylab = "-Log(P-Value)")
abline(v = c(-2,2), col = "gray", lty = 2)
abline(h = -log(0.1), col = "gray", lty = 2)
```

